server/src/main/java/com/cloud/storage/snapshot/SnapshotSchedulerImpl.java

```diff
@@ -378,6 +378,12 @@ public Date scheduleNextSnapshotJob(final SnapshotPolicyVO policy) {
         if (policyId == Snapshot.MANUAL_POLICY_ID) {
             return null;
         }
+
+        if (_volsDao.findById(policy.getVolumeId()) == null) {
+            s_logger.warn("Found snapshot policy ID: " + policyId + " for volume ID: " + policy.getVolumeId() + " that does not exist or has been removed");
+            return null;
+        }
```

Comment:
great find, should we also delete the snapshotpolicy while we are at it?

Reply:
Makes sense

Reply:
Added code to remove schedule when volume is missing